llama_index/storage/storage_context.py

```diff
@@ -100,18 +101,19 @@ def persist(
 
         Args:
             persist_dir (str): directory to persist the storage context
-
         """
         if fs is not None:
+            persist_dir = str(persist_dir)  # NOTE: doesn't support Windows here
             docstore_path = concat_dirs(persist_dir, docstore_fname)
             index_store_path = concat_dirs(persist_dir, index_store_fname)
             vector_store_path = concat_dirs(persist_dir, vector_store_fname)
             graph_store_path = concat_dirs(persist_dir, graph_store_fname)
         else:
-            docstore_path = str(Path(persist_dir) / docstore_fname)
-            index_store_path = str(Path(persist_dir) / index_store_fname)
-            vector_store_path = str(Path(persist_dir) / vector_store_fname)
-            graph_store_path = str(Path(persist_dir) / graph_store_fname)
+            persist_dir = Path(persist_dir)
+            docstore_path = str(persist_dir / docstore_fname)
```

Comment:
I think this will raise an error without `Path(persist_dir)` -- we can't divide strings :)

Reply:
I think it's a `Path` in this flow though, not `str`. I am mobile rn, so my apologies if wrong

Reply:
oh woops lol definitely misread that. Yea that works